src/System.Drawing.Common/src/System/Drawing/Imaging/ImageFormat.cs

```diff
@@ -161,16 +161,16 @@ internal ImageCodecInfo FindEncoder()
         /// </summary>
         public override string ToString()
         {
-            if (this == s_memoryBMP) return "MemoryBMP";
-            if (this == s_bmp) return "Bmp";
-            if (this == s_emf) return "Emf";
-            if (this == s_wmf) return "Wmf";
-            if (this == s_gif) return "Gif";
-            if (this == s_jpeg) return "Jpeg";
-            if (this == s_png) return "Png";
-            if (this == s_tiff) return "Tiff";
-            if (this == s_exif) return "Exif";
-            if (this == s_icon) return "Icon";
+            if (this.Guid == s_memoryBMP.Guid) return "MemoryBMP";
+            if (this.Guid == s_bmp.Guid) return "Bmp";
+            if (this.Guid == s_emf.Guid) return "Emf";
+            if (this.Guid == s_wmf.Guid) return "Wmf";
+            if (this.Guid == s_gif.Guid) return "Gif";
+            if (this.Guid == s_jpeg.Guid) return "Jpeg";
+            if (this.Guid == s_png.Guid) return "Png";
+            if (this.Guid == s_tiff.Guid) return "Tiff";
+            if (this.Guid == s_exif.Guid) return "Exif";
+            if (this.Guid == s_icon.Guid) return "Icon";
             return "[ImageFormat: " + _guid + "]";
```

Comment:
Unrelated to this PR, but @safern, you were asking about finding places where the new span-based support and String.Create can be used to avoid allocations.  From a perf standpoint, it probably doesn't matter at all, as I doubt using ToString on an ImageFormat is on anyone's hot path, and as such it's not worth the complication, but just as an example, this is an example where allocations could be reduced.  If you were to look at what this compiles down to, it'll end up being a `string.Concat(object, object, object)` call, which means the `Guid` needs to be boxed (one allocation), then ToString will be called on it (a second allocation), and then finally the result string will be created (third allocation).  Those first two can be avoided, e.g. something like:
```C#
return string.Create(14 + 36 + 1, _guid, (span, state) =>
{
    "[ImageFormat: ".AsSpan().CopyTo(span);
    state.TryFormat(span.Slice(14), out _);
    span[span.Length - 1] = ']';
});
```

Reply:
Awesome. Will include it just because it is a nice to have even though it is not a hot path. Thanks for the hint! :)

Reply:
Changing this to the Span version adds negative value because of it is larger code. The larger code does not pay for itself here.

Reply:
Ok, so, making larger code for this kind of improvements only pays off if it is a hot path?

Reply:
Right.

Reply:
Ok, thanks!

Reply:
Right. I was just pointing it out as a pattern, but as noted not actually to be changed here.

Reply:
There is another problem here: there is no pint in doing multiple `if - return` checks since this will increase processing time significantly. Most efficient and clean version would be based on `switch` clause with check of chosen 8 bytes  from 16 byte guids.